README.md

```diff
@@ -38,91 +32,300 @@ IncludeOS is free software, with "no warranties or restrictions of any kind".
+and generates the required virtualenv scripts and cmake information.
+
+Next to build the service do:
+
+```
+  $ cmake ..
+  $ cmake --build .
+
+```
+Doing `cmake` configures and generates the build files and they are written to
+the build folder. Then doing `cmake --build .` builds the target service. You
+should see the last line as:
+
+```
+  [100%] Built target demo
+```
+`demo` is the name of this service for the demo_service. That's the name you will
+use when starting the service. Do `ls` to see all the files created. There is also
+an executable file named `demo` which is now the service you will start.
+
+
+##### Starting a service
+
+Now that you have all your build requirements ready, you can start the service.
+
+However to start a service a good practice is to always activate the service
+requirements;
+
+```
+  $ source activate.sh
+```
+
+Now you can boot the demo service with `boot <service-name>`.
+
+```
+  $ boot demo
+```
+
+This should start your demo service and show something along the following lines:
+
+```
+  ================================================================================
+  IncludeOS  (x86_64 / 64-bit)
+  +--> Running [ IncludeOS Demo Service ]
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  Service started
+  Made device_ptr, adding to machine
+    [ Network ] Creating stack for VirtioNet on eth0 (MTU=1500)
+       [ Inet ] Bringing up eth0 on CPU 0
+  *** Basic demo service started ***
+```
+
+To build more examples follow the [README](https://github.com/includeos/demo-examples/blob/master/README.md) in the demo-examples repository.
+
+##### Writing your first service
+
+To start writing your first service,
+- Copy the [demo_service/service.cpp](https://github.com/includeos/demo-examples/blob/master/demo_service/service.cpp) example.
+- Then start implementing the `Service::start` function in the `Service` class, located in `<your-service-name>/service.cpp` (very simple example provided). This function will be called once the OS is up and running.
+- Update the `CMakeLists.txt` to specify the name of your project, enable any needed drivers or plugins, etc.
+- Update the `conanfile.txt` to specify any build requirements and generators.
+
+You should then be able to run your service in the same way as the demo_service.
+
+To add a new service to our demo examples, make a PR to our [demo-examples](https://github.com/includeos/demo-examples) repo. Make sure to
+add a README in your example folder with description of your service.
+
+**Note:** Remember to deactivate your service environment after your work with:
+
+```
+  $ source deactivate.sh
+```
+
+<!-- TODO: How building with Conan works? -->
+
+### Getting started developing packages
+
+Currently building works for `clang-6` and `gcc-7.3.0` compiler toolchain.
+It is expected that these are already installed in your system. However we
+hope to provide toolchains in the future.
```

Comment:
"compiler toolchains". 
                   plural --^

"We hope to provide toolchains in the future" - I'd drop that